util/design/secded_gen.py

```diff
@@ -392,13 +394,30 @@ def ecc_encode(codetype: str, k: int, dataword: int) -> Tuple[int, int]:
         # Add ECC bit inversion if needed (see print_enc function).
         bit ^= (invert & k % 2)
         codeword = str(bit) + codeword
+    return codeword
+
+
+def ecc_encode(codetype: str, k: int, dataword: int) -> Tuple[int, int]:
+    log.info(f"Encoding ECC for {hex(dataword)}")
+
+    m, bitmasks, invert = _ecc_pick_code(codetype, k)
+    codeword = _ecc_encode(k, m, bitmasks, invert, dataword)
 
     # Debug printouts
     log.debug(f'original hex: {hex(dataword)}')
     log.debug(f'codeword hex: {hex(int(codeword,2))}')
     return int(codeword, 2), m
 
 
+def ecc_encode_some(codetype: str,
+                    k: int,
+                    datawords: int) -> Tuple[List[int], int]:
+    m, bitmasks, invert = _ecc_pick_code(codetype, k)
+    codewords = [int(_ecc_encode(k, m, bitmasks, invert, w), 2)
```

Comment:
just a nit, does this technically mean the type for `datawords` should no longer be int? But a list of ints?

Reply:
Yes it does! Oops! (A reason that I don't tend to use type annotations unless I've got some tool checking them: they're just too easy to get wrong!)

Reply:
yeah completely agreed. I have a love hate relationship with the mypy checks on reggen. 
On one hand its like "gahhh! just let me merge!" but on the other hand it really helps clarify things...